bounties_api/user/middleware.py

```diff
@@ -1,10 +1,20 @@
 from user.backend import get_user
 from django.utils.deprecation import MiddlewareMixin
+from django.conf import settings
+import jwt
 
 
 class AuthenticationMiddleware(MiddlewareMixin):
     def process_request(self, request):
         public_address = request.session.get('public_address', None)
+        if not public_address:
+            jwt_token = request.META.get('HTTP_AUTHENTICATION', '')
+            try:
+                payload = jwt.decode(jwt_token, settings.SECRET_KEY)
+                public_address = payload.get('public_address', '')
+            except:
+                pass
```

Comment:
Are there any types of exceptions we're interested in here, for our own error tracking? Catching exceptions where there is no token makes sense, but an issue with how we're handling tokens might be lost in this catch-all pass.

Yet, all we're really doing is decoding the token with our `settings.SECRET_KEY`, and it's hard to imagine anything going wrong. Still, to harden this for future changes, being specific about what we catch would let us at least hear about new issues that are unexpected.

Reply:
There is an expiration exception - this will be after the 2 week period. We can make this more sophisticated later.